crypto/ui/ui_util.c

```diff
@@ -104,12 +104,13 @@ static int ui_read(UI *ui, UI_STRING *uis)
     switch (UI_get_string_type(uis)) {
     case UIT_PROMPT:
         {
-            char result[PEM_BUFSIZE + 1];
+            char result[PEM_BUFSIZE];
             const struct pem_password_cb_data *data =
                 UI_method_get_ex_data(UI_get_method(ui), ui_method_data_index);
             int maxsize = UI_get_result_maxsize(uis);
             int len = data->cb(result,
-                               maxsize > PEM_BUFSIZE ? PEM_BUFSIZE : maxsize,
+                               maxsize + 1 > PEM_BUFSIZE ? PEM_BUFSIZE
+                                                         : maxsize + 1,
```

Comment:
The docs here (in `PEM_read_bio_PrivateKey.pod`) are seemingly ambiguous, saying `I<size> is the maximum length of the passphrase (i.e. the size of buf)`, but a "length of the passphrase" seems like it would not be expected to count for the trailing NUL, where "size of buf" clearly must account for the trailing NUL.  That said, there are later segments in the doc, like "makes no assumption regarding the pass phrase received from the password callback.  It will simply be treated as a byte sequence", that suggest that the trailing NUL should be accounted for.

Reply:
No, if you count NUL into the len returned it means the NUL byte is part of the passphrase. Which it usually isn't. And yeah we should clarify the size meaning size of the buf and not the length of the passphrase.